bblocks/import_tools/world_bank_projects.py

```diff
@@ -0,0 +1,559 @@
+    of the data to import. If no dates are specified, all data will be imported.
+    To import the data, call the load_data method. If the data has already downloaded, it will
+    be loaded to the object from disk, otherwise it will be downloaded from the API.
+    To retrieve the data, call the get_data method. You can specify the type of data to retrieve,
+    either 'general' or 'theme'. If no type is specified, 'general' data will be returned.
+    To update the data, call the update_data method. This will download the data from the API.
+    If 'reload' is set to True, the data will be reloaded to the object.
+
+    Parameters:
+        start_date: start date of data to import, in YYYY-MM-DD format
+        end_date: end date of data to import, in YYYY-MM-DD format.
+    """
+
+    start_date: str | None = None
+    end_date: str | None = None
+
+    @property
+    def _path(self):
+        """Generate path based on version"""
+
+        start_date = f"_{self.start_date}" if self.start_date is not None else ""
+        end_date = f"_{self.end_date}" if self.end_date is not None else ""
+
+        return BBPaths.raw_data / f"world_bank_projects{start_date}{end_date}.json"
+
+    def _format_general_data(self, additional_fields: list = None) -> None:
+        """Clean and format general data and store it in _data attribute with key 'general_data'"""
+
+        numeric_cols = [
+            "lendprojectcost",
+            "totalcommamt",
+            "grantamt",
+            "idacommamt",
+            "ibrdcommamt",
+            "curr_total_commitment",
+            "curr_ibrd_commitment",
+            "curr_ida_commitment",
+            "curr_project_cost",
+        ]
+
+        self._data["general_data"] = (
+            pd.DataFrame.from_dict(self._raw_data, orient="index")
+            .reset_index(drop=True)
+            .filter(list(GENERAL_FIELDS) + additional_fields, axis=1)
+            # change the fiscal year to int
+            .assign(
+                approvalfy=lambda d: clean.clean_numeric_series(d["approvalfy"], to=int)
+            )
+            # change numeric columns to float
+            .pipe(clean.clean_numeric_series, series_columns=numeric_cols)
+            .assign(  # format dates
+                boardapprovaldate=lambda d: clean.to_date_column(
+                    d["boardapprovaldate"]
+                ),
+                closingdate=lambda d: clean.to_date_column(d["closingdate"]),
+                p2a_updated_date=lambda d: clean.to_date_column(d["p2a_updated_date"]),
+            )
+            # rename columns
+            .rename(columns=GENERAL_FIELDS)
+        )
+
+    def _format_theme_data(self) -> None:
+        """Format theme data and store it as a dataframe in _data attribute with key 'theme_data'"""
+
+        theme_data = []
+        for _, proj_data in self._raw_data.items():
+            theme_data.extend(clean_theme(proj_data))
+
+        self._data["theme_data"] = pd.DataFrame(theme_data)
```

Comment:
I think this should be ordered so that the columns appear as:
`project ID` 
`theme1`
`theme2`
`theme3`
`percent`

Otherwise, the output feels a confusing to me. (I know we've discussed my confusion at length, but this feels like a fairly easy way to make it a bit clearer